### PR TITLE
optimised stream handling in CatalogServiceHandler

### PR DESCRIPTION
**Pull request by rjayasinghe**

### CODE REVIEW

File: srv/src/main/java/my/bookshop/handlers/CatalogServiceHandler.java [hunk 1]
```diff
@@ -34,14 +35,23 @@ public CatalogServiceHandler(PersistenceService db) {
...
 	}
 
+	private void discountBooksWithMoreThan111Stock(Books b) {
+		if(b.getStock() > 111) {
+			b.setTitle(String.format("%s -- 11%% discount", b.getTitle()));
+		}
+	}
+
+	private void loadStockIfNotSet(Books b) {
+		if (b.getStock()== null) {
```
Comment (beckermarc) [inline srv/src/main/java/my/bookshop/handlers/CatalogServiceHandler.java:53]:
```suggestion
		if (b.getStock() == null) {
```

**Review by beckermarc — APPROVED** [review]

File: srv/src/main/java/my/bookshop/handlers/CatalogServiceHandler.java (hunk 1 above)
Comment (rjayasinghe) [reply srv/src/main/java/my/bookshop/handlers/CatalogServiceHandler.java:53]: Obviously, I have no code formatting rules installed, yet. ;)

Comment (beckermarc) [reply srv/src/main/java/my/bookshop/handlers/CatalogServiceHandler.java:53]: We also didn't strictly define any so far for this project.. ;)

**Comment by rjayasinghe** [comment]
The last commit only added a whitespace. Can I see the test execution from the first commit in this PR?

File: srv/src/main/java/my/bookshop/handlers/CatalogServiceHandler.java (hunk 1 above)
Comment (rjayasinghe) [reply srv/src/main/java/my/bookshop/handlers/CatalogServiceHandler.java]: I corrected the whitespace but now the build fails. 🤯 😄 

I tried the build on my Mac (maven 3.6 and jdk 8) and its runs without failure.

**Comment by beckermarc** [comment]
```
Failed to execute goal org.apache.maven.plugins:maven-jar-plugin:2.4:jar (default-jar) on project bookshop: Execution default-jar of goal org.apache.maven.plugins:maven-jar-plugin:2.4:jar failed: Plugin org.apache.maven.plugins:maven-jar-plugin:2.4 or one of its dependencies could not be resolved: Could not transfer artifact org.codehaus.plexus:plexus-interpolation:jar:1.15 from/to central (https://repo.maven.apache.org/maven2): Transfer failed for https://repo.maven.apache.org/maven2/org/codehaus/plexus/plexus-interpolation/1.15/plexus-interpolation-1.15.jar: Connection reset -> [Help 1]
```

This seems to be a network issue. I triggered a rebuild.

**Pull request merged**